src/poetry/packages/locker.py

```diff
@@ -463,15 +449,7 @@ def _get_content_hash(self) -> str:
 
         return content_hash
 
-    def _get_lock_data(self) -> "TOMLDocument":
-        if not self._lock.exists():
-            raise RuntimeError("No lockfile found. Unable to read locked packages")
-
-        try:
-            lock_data = self._lock.read()
-        except TOMLKitError as e:
-            raise RuntimeError(f"Unable to read the lock file ({e}).")
-
+    def _verify_lock_data(self, lock_data: dict) -> NoReturn:
```

Comment:
`NoReturn` is not correct -- it signifies a method that will never return (i.e. does not terminate, always throws an exception, etc). If there is no `return` statement the correct type is still `None` (as Python will add an implicit return at the end of the function).